src/lib.rs

```diff
@@ -2033,4 +2179,117 @@ mod test {
         assert_eq!(products[0], products[1]);
         assert_eq!(products[0], products[2]);
     }
+
+    mod formatter {
+        #[cfg(feature = "std")]
+        #[test]
+        fn format_as_decimal() {
+            use super::{Ratio, Zero};
+
+            let data: [(u8, u8, &'static str, bool); 60] = [
+                (1u8, 2u8, "0.5", true),
+
+                (1u8, 255u8, "0.0039215686274509803921568627450980392156862745098039215686274509", false),
+                (1u8, 254u8, "0.0039370078740157480314960629921259842519685039370078740157480314", false),
+                (1u8, 253u8, "0.003952569169960474308300395256916996047430830039525691699604743", false),
+                (1u8, 252u8, "0.0039682539682539682539682539682539682539682539682539682539682539", false),
+                (1u8, 251u8, "0.0039840637450199203187250996015936254980079681274900398406374501", false),
+                (1u8, 250u8, "0.004", true),
+                (1u8, 112u8, "0.0089285714285714285714285714285714285714285714285714285714285714", false),
+                (1u8, 111u8, "0.009009009009009009009009009009009009009009009009009009009009009", false),
+                (1u8, 96u8, "0.0104166666666666666666666666666666666666666666666666666666666666", false),
+                (1u8, 92u8, "0.0108695652173913043478260869565217391304347826086956521739130434", false),
+                (1u8, 91u8, "0.0109890109890109890109890109890109890109890109890109890109890109", false),
+                (1u8, 90u8, "0.0111111111111111111111111111111111111111111111111111111111111111", false),
+                (1u8, 9u8, "0.1111111111111111111111111111111111111111111111111111111111111111", false),
+                (1u8, 8u8, "0.125", true),
+                (1u8, 7u8, "0.1428571428571428571428571428571428571428571428571428571428571428", false),
+                (1u8, 6u8, "0.1666666666666666666666666666666666666666666666666666666666666666", false),
```

Comment:
We ought to be rounding values -- "0.166...667"

... which I realize is harder, but I think it's important.